### DISCUSSION TITLE
running same requests for `n` time?

### DISCUSSION THREAD

**Discussion by dibalikpohon** [Orange-OpenSource/hurl#1870] [Ideas] [answered]
Hi!

I am willing to have sequence of test scenarios like this:
1. Setup. Run multiple same `POST` requests with different payload. This request adds data to the table.
2. Run tests. The test can be anything, including adding and deleting data.
3. Teardown. Run multiple `DELETE` request with different path variables. This deletes all the data from the table.

I am okay with number 1 and number 2, but

I was thinking that I could capture id for each inside array like
```
GET http://localhost:8082/api/data

HTTP 200
[Captures]
ids: jsonpath "$.data[*].id"
```
so I can use it like this
```
DELETE http://localhost:8082/api/data/{{ids...}}

HTTP 200
```

This runs `DELETE` request for each `ids` in an array. Kind of like [Spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) in JavaScript.

I appreciate any discussion here. Thank you.

**Comment by fabricereix** [comment, marked as answer]
we are going to add the `repeat` option #88, to repeat the same request n times

to inject different variables for each request, we could use the index of the request in the loop/repeat

Implicitly
```
DELETE http://localhost:8082/api/data/{{ids nth repeat_index}}
[Options]
repeat: 5

HTTP 200
```

or explicitly 
```
DELETE http://localhost:8082/api/data/{{id}}
[Options]
for: id in ids

HTTP 200
```

I think I would prefer the explicit version. It may be a bit weird to have that in the `[Option]` section.
for sure, we will implement the simple `repeat` option.

**Comment by jcamiel** [comment]
@dibalikpohon `--repeat` option has been implemented in Hurl 5.0.0, either by command line

```shell
$ hurl --repeat 2 a.hurl b.hurl
```
will run `a.hurl` `b.hurl` `a.hurl` `b.hurl`

Or on a specific request:

```
GET https://foo.com
HTTP 200

GET https://bar.com
[Options]
repeat: 10
HTTP 200
```

**Reply by gaetan-osrd** [reply]
I tried this
```
GET https://api.restful-api.dev/objects
HTTP 200
[Captures]
ids: jsonpath "$[*].id"
number: jsonpath "$[*].id" count
GET https://api.restful-api.dev/object/{{ ids nth repeat_count }}
[Options]
repeat: {{ number }}
HTTP 200
``` 
but it failed with 
```
error: Unrenderable expression
  --> coffees.hurl:6:43
   |
 6 | GET https://api.restful-api.dev/object/{{ ids nth repeat_count }}
```
Did I miss something ? The ultimate goal is to test batch upload worked fine by getting extra properties from each uploaded entries

Edit anyway repeat_count is not a variable

**Reply by jcamiel** [reply]
Hi @gaetan-osrd 

This syntax `{{ ids nth repeat_count }}` is not yet supported by Hurl. Not sure how to adress it but if your APIs respect some pagination or HATEOAS style you can update a "next" id or next URL to be updated within the repeat loop:

```hurl
GET https://api.restful-api.dev/objects
HTTP 200
[Captures]
ids: jsonpath "$[*].id"
number: jsonpath "$[*].id" count
id: ids first

GET https://api.restful-api.dev/object/{{id}}
[Options]
repeat: {{ number }}
HTTP 200
[Captures]
id: jsonpath "$.next.id"
```

In this sample, the variable id is updated for each loop

**Reply by gaetan-osrd** [reply]
Unfortunately I don't have the notion of sequences. I guess the idea of the unrenderable things shoudl be to evaluate the whole expression and check the type like it's done in captures. I might give it a try at a moment if I should need it later
